src/mergely.js

```diff
@@ -1,1776 +1,92 @@
-"use strict";
+const CodeMirrorDiffView = require('./diff-view');
 
-require('./mergely.css');
-
-(function(jQuery, CodeMirror) {
-
-var Mgly = {};
-
-Mgly.Timer = function(){
-	var self = this;
-	self.start = function() { self.t0 = new Date().getTime(); };
-	self.stop = function() {
-		var t1 = new Date().getTime();
-		var d = t1 - self.t0;
-		self.t0 = t1;
-		return d;
-	};
-	self.start();
-};
-
-Mgly.ChangeExpression = new RegExp(/(^(?![><\-])*\d+(?:,\d+)?)([acd])(\d+(?:,\d+)?)/);
-
-Mgly.DiffParser = function(diff) {
-	var changes = [];
-	var change_id = 0;
-	// parse diff
-	var diff_lines = diff.split(/\n/);
-	for (var i = 0; i < diff_lines.length; ++i) {
-		if (diff_lines[i].length == 0) continue;
-		var change = {};
-		var test = Mgly.ChangeExpression.exec(diff_lines[i]);
-		if (test == null) continue;
-		// lines are zero-based
-		var fr = test[1].split(',');
-		change['lhs-line-from'] = fr[0] - 1;
-		if (fr.length == 1) change['lhs-line-to'] = fr[0] - 1;
-		else change['lhs-line-to'] = fr[1] - 1;
-		var to = test[3].split(',');
-		change['rhs-line-from'] = to[0] - 1;
-		if (to.length == 1) change['rhs-line-to'] = to[0] - 1;
-		else change['rhs-line-to'] = to[1] - 1;
-		change['op'] = test[2];
-		changes[change_id++] = change;
-	}
-	return changes;
-};
-
-Mgly.sizeOf = function(obj) {
-	var size = 0, key;
-	for (key in obj) {
-		if (obj.hasOwnProperty(key)) size++;
+class Mergely {
+	constructor(element, options) {
+		this._diffView = new CodeMirrorDiffView(element, options, {
+			CodeMirror
```

Comment:
I suggest adding a mandatory import of CodeMirror:
`const CodeMirror = require('codemirror');`
Since without this dependency Mergely does not work independently.